django_mongodb/indexes.py

```diff
@@ -0,0 +1,76 @@
+from django.db import NotSupportedError
+from django.db.models import Index
+from django.db.models.fields.related_lookups import In
+from django.db.models.lookups import BuiltinLookup
+from django.db.models.sql.query import Query
+from django.db.models.sql.where import AND, XOR, WhereNode
+
+from .query_utils import process_rhs
+
+MONGO_INDEX_OPERATORS = {
+    "exact": "$eq",
+    "gt": "$gt",
+    "gte": "$gte",
+    "lt": "$lt",
+    "lte": "$lte",
+    "in": "$in",
+}
+
+
+def _get_condition_mql(self, model, schema_editor):
+    """Analogous to Index._get_condition_sql()."""
+    query = Query(model=model, alias_cols=False)
+    where = query.build_where(self.condition)
+    compiler = query.get_compiler(connection=schema_editor.connection)
+    return where.as_mql_idx(compiler, schema_editor.connection)
+
+
+def builtin_lookup_idx(self, compiler, connection):
+    lhs_mql = self.lhs.target.column
+    value = process_rhs(self, compiler, connection)
+    try:
+        operator = MONGO_INDEX_OPERATORS[self.lookup_name]
+    except KeyError:
+        raise NotSupportedError(
+            f"MongoDB does not support the '{self.lookup_name}' lookup in indexes."
+        ) from None
+    return {lhs_mql: {operator: value}}
+
+
+def in_idx(self, compiler, connection):
+    if not connection.features.supports_in_index_operator:
+        raise NotSupportedError(
+            f"MongoDB < 6.0 does not support the '{self.lookup_name}' lookup in indexes."
```

Comment:
Couldn't this just be:
```suggestion
            f"MongoDB < 6.0 does not support `in` lookup in indexes."
```